d2/src/main/java/com/linkedin/d2/balancer/servers/ZooKeeperAnnouncer.java

```diff
@@ -130,6 +167,12 @@ synchronized void retry(Callback<None> callback)
       {
         markDown(callback);
       }
+
+      // If markDown for warm-up cluster is pending, complete it
+      if(!_pendingWarmUpMarkDown.isEmpty() && _isWarmUp)
+      {
+        _server.markDown(_warmupClusterName, _uri, callback);
+      }
```

Comment:
May want to move this to the top, above the check for `if (_isUp)`. Having it at the end seems like it could short-circuit the warmup that was initialized from the retried markUp.

Reply:
thanks! Fixed it.